buildpacks/python.go

```diff
@@ -33,119 +32,120 @@ func (bt PythonBuildTool) Version() string {
-		fmt.Printf("Virtualenv binary already installed in %s\n", virtualenvBin)
-	} else {
-		fmt.Printf("Installing virtualenv for Python in %s\n", pythonVersionDir)
+	return nil
+}
 
-		shimsDir := filepath.Join(pyenvDir, "shims")
+func (bt PythonBuildTool) DownloadUrl() string {
+	opsys := OS()
+	arch := Arch()
+	extension := "sh"
+	version := bt.Version()
 
-		os.Setenv("PYENV_ROOT", pyenvDir)
-		os.Setenv("PYENV_SHELL", "sh")
-		os.Setenv("PYENV_VERSION", bt.Version())
+	if version == "" {
+		version = "latest"
+	}
 
-		PrependToPath(filepath.Join(pyenvDir, "bin"))
-		PrependToPath(shimsDir)
-		setupCmd := fmt.Sprintf("pyenv rehash")
-		ExecToStdout(setupCmd, pyenvDir)
+	if arch == "amd64" {
+		arch = "x86_64"
+	}
 
-		cmd := "pip install virtualenv"
-		ExecToStdout(cmd, pyenvDir)
+	if opsys == "darwin" {
+		opsys = "MacOSX"
 	}
 
-	if _, err := os.Stat(virtualenvDir); err == nil {
-		fmt.Printf("Virtualenv for %s exists in %s\n", bt.Version(), virtualenvDir)
-	} else {
-		fmt.Println("Creating virtualenv...")
-		MkdirAsNeeded(virtualenvDir)
-		pythonBinPath := filepath.Join(pythonVersionDir, "bin", "python")
-		cmd := fmt.Sprintf("%s -p %s %s", virtualenvBin, pythonBinPath, virtualenvDir)
-		ExecToStdout(cmd, bt.spec.PackageDir)
+	if opsys == "linux" {
+		opsys = "Linux"
 	}
 
-	return nil
-}
+	if opsys == "windows" {
+		opsys = "Windows"
+		extension = "exe"
+	}
 
-func (bt PythonBuildTool) Setup() error {
-	virtualenvDir := bt.pkgVirtualEnvDir()
-	PrependToPath(filepath.Join(virtualenvDir, "bin"))
+	data := struct {
+		PyNum     int
+		OS        string
+		Arch      string
+		Version   string
+		Extension string
+	}{
+		3,
+		opsys,
+		arch,
+		AnacondaToolVersion,
+		extension,
+	}
 
-	return nil
+	url, _ := TemplateToString(ANACONDA_URL_TEMPLATE, data)
 
+	return url
 }
 
-func (bt PythonBuildTool) InstallPlatformDependencies() error {
-	gi := goInfo.GetInfo()
-	if gi.GoOS == "darwin" {
-		if strings.HasPrefix(gi.Core, "18.") {
-			// Need to install the headers on Mojave
-			if !PathExists("/usr/include/zlib.h") {
-				installCmd := "sudo -S installer -pkg /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg -target /"
-				fmt.Println("Going to run:", installCmd)
-				cmdArgs := strings.Split(installCmd, " ")
-				cmd := exec.Command(cmdArgs[0], cmdArgs[1:]...)
-				cmd.Stdout = os.Stdout
-				cmd.Stdin = os.Stdin
-				cmd.Stderr = os.Stderr
-				cmd.Run()
+func (bt PythonBuildTool) Setup() error {
+	condaDir := bt.AnacondaInstallDir()
+	envDir := bt.EnvironmentDir()
+
+	if _, err := os.Stat(envDir); err == nil {
+		fmt.Printf("environment installed in %s\n", envDir)
+	} else {
+		currentPath := os.Getenv("PATH")
+		newPath := fmt.Sprintf("PATH=%s:%s", filepath.Join(condaDir, "bin"), currentPath)
+		setupDir := bt.spec.PackageDir
+		condaBin := filepath.Join(condaDir, "bin", "conda")
+
+		for _, cmd := range []string{
+			fmt.Sprintf("%s config --set always_yes yes --set changeps1 no", condaBin),
```

Comment:
I'm not sure if this commands should be used every time to setup Anaconda/Miniconda

Reply:
It can't hurt; it does ensure that the config is properly set

Reply:
Miniconda does provide a python binary, we then install the requested python in an environment and use that.